lib/calendar_interval.ex

```diff
@@ -714,11 +842,16 @@ defmodule CalendarInterval do
          NaiveDateTime.compare(ndt, last) in [:eq, :lt]}
     end
 
-    def member?(interval, %Date{} = date) do
+    def member?(interval, %Date{calendar: Calendar.ISO} = date) do
       {:ok, ndt} = NaiveDateTime.new(date, ~T"00:00:00")
       member?(interval, ndt)
     end
 
+    def member?(interval, %Date{calendar: calendar} = date) do
```

Comment:
Thats interesting and I hadn't thought about it. In Allen's interval terms its asking the question of whether one interval `starts`, `finishes`, `during` or `equals` the other. Maybe that's the semantics that could be applied?  

In your first example its striating the first interval as a scalar, in the second more like a vector. I think it should always be considered a vector - and hence using Allen's Interval semantics.  But its only my first coffee so maybe not fully thought out.

Reply:
Just to be clear, what I meant was that my understanding of Elixir enumerable semantics is that: if element X is _in_ enumerable XS then if we list all elements of XS, X will be one of them (as in `===/2`) (and pardon for my bad attempt at trying to come up with a property :))

So, taking the example from comment above, we'd have the following:

```elixir
iex> ~I"2019-01" in ~I"2019-01/12"
true

iex> ~I"2019-01-01" in ~I"2019-01/12"
false

iex> CalendarInterval.enclosing(~I"2019-01-01", :month) in ~I"2019-01/12"
true

iex> CalendarInterval.<some_predicate>?(~I"2019-01/12", ~I"2019-01-01")
true
```

I have hard time coming up with <some_predicate> function name, perhaps a generic `includes?` isn't too bad.

or maybe it'd be enough for people to do:

```
iex> CalendarInterval.relation(~I"2019-12-01", ~I"2019-01/12") in [:starts, :during, :finishes]
true
```

(it currently does not work because we can only calculate relation of intervals of the same precision, but I think we can loosen it up a bit)

In any case, first of all, sorry for dragging you into this discussion with myself :D Second of all, does the concern make sense? And what do you think about the changes to the API?

Reply:
I guess the core problem here is that `~I"2019-01" in ~I"2019-01/12"` is not "element in enumerable", but rather "interval contained in interval". It becomes even more tricky when using an example, where the concept of "element in enumerable" breaks down: 
`~I"2019-01" in ~I"2019-01-01/20"`
Would that result in `true` or `false`? If get's easier when we declare "x in y" returns if x is contained in y, for the ease of it's api, but it's certainly different semantics than "element in enumerable". The elixir core equivalent would actually be:
```
> 1..5 in 1..10
false
```

Reply:
I guess sticking to `x in y` results in true if `x` is a subinterval of `y` with the same granularity of `y` is the most sane approach, for the reasons mentioned initially:
```
 > Enum.into(~I"2019-01/12", [])
[~I"2019-01", ~I"2019-02", ~I"2019-03", …]
```
There's no other sane return value for the above if a CalendarInterval by itself should be an `Enumerable` (and imho it should be). We have no good other ways to denote "month" or "years" in elixir without falling back to strings or using e.g. the first date of the greater granularity value.

Reply:
> I guess sticking to x in y results in true **if** x is a subinterval of y with the **same granularity** of y is the most sane approach

(emphasis mine) yes, that's a good way to put it, I'm leaning this way too. Thanks for feedback!